graf2d/gpadv7/v7/inc/ROOT/TPad.hxx

```diff
@@ -131,10 +132,27 @@ public:
       fPrimitives.clear();
    }
 
-   void CreateFrame();
+   void CreateFrameIfNeeded();
 
    const TFrame *GetFrame() const { return fFrame.get(); }
 
+   TPadUserAxisBase* GetAxis(size_t dimension) const;
+   TPadUserAxisBase* GetExistingAxis(size_t dimension) const;
+
+   void SetAxisBounds(int dimension, double begin, double end);
+   void SetAxisBound(int dimension, TPadUserAxisBase::EAxisBoundsKind boundsKind, double bound);
+   void SetAxisAutoBounds(int dimension);
+
+   void SetAllAxisBounds(const std::vector<std::array<double, 2>> &vecBeginAndEnd);
+
+   /// Simple struct representing an axis bound.
+   struct BoundKindAndValue {
```

Comment:
Is this part of the public interface?

Reply:
Yes. It allows you to specify `SetAllAxisBound({{10., kAxisBoundsEnd}, {0., kAxisBoundsBegin}})`

Reply:
Is the default constructor setting the double value to a reasonable value for this purpose?

Also it looks like the code you shown would not compile (is the order of the member correct)?